slices/ca-certificates.yaml

```diff
@@ -0,0 +1,66 @@
+package: ca-certificates
+
+essential:
+  - ca-certificates_copyright
+
+slices:
+  # This slice only contains the certificates bundle at /etc/ssl/certs/ca-certificates.crt.
+  # To keep the individual certificates at /usr/share/ca-certificates/mozilla/ or the
+  # configuration file at /etc/ca-certificates.conf, please use the "data-with-certs" slice.
+  # Please note that the mutation script (partially) mimics the deb's maintainer
+  # scripts to generate the certificates bundle.
+  data:
+    essential:
+      # This adds the symlink at etc/ssl/certs.pem for OpenSSL to look up the
+      # bundled certificate file. See "openssl_data" for details.
+      - openssl_data
+    contents:
+      /etc/ssl/certs/ca-certificates.crt: {text: FIXME, mutable: true}
+      /usr/share/ca-certificates/mozilla/: {until: mutate}
+      /usr/share/ca-certificates/mozilla/**: {until: mutate}
+    mutate: |
+      certs_dir = "/usr/share/ca-certificates/mozilla/"
+      certs = [
+        content.read(certs_dir + path) for path in content.list(certs_dir)
+      ]
+      content.write("/etc/ssl/certs/ca-certificates.crt", "".join(certs))
+
+  # This slice offers a standard chunk of the ca-certificates package.
+  # It contains the default certificates at /usr/share/ca-certificates/mozilla,
+  # a certificates bundle at /etc/ssl/certs/ca-certificates.crt and a config
+  # file at /etc/ca-certificates.conf.
+  data-with-certs:
+    essential:
+      - ca-certificates_data
+    contents:
+      # This config file is generated by the deb's maintainer scripts.
+      /etc/ca-certificates.conf: {text: FIXME, mutable: true}
+      /usr/share/ca-certificates/mozilla/**:
+    mutate: |
+      certs_dir = "/usr/share/ca-certificates/mozilla/"
+      names = [
+        "mozilla/{}".format(name) for name in content.list(certs_dir)
+      ]
+      content.write("/etc/ca-certificates.conf", "\n".join(names))
+
+  bins:
+    essential:
+      - base-files_bin
+      - base-files_tmp
+      # Manually linking /bin/sh to /bin/bash is required as /bin/sh is hardcoded in
+      # the shebang of the script.
+      # See https://github.com/canonical/chisel-releases/pull/61#discussion_r1326873419
+      - bash_bins
+      - coreutils_bins
+      - findutils_bins
+      - openssl_bins
+      - sed_bins
```

Comment:
can we do the same analysis as in #338 as split some of these binary slices into more granular ones?

Reply:
We can do that. But it's quite a number of binaries used here in comparison to #338. The `update-ca-certificates` script needs:

- rm
- mktemp
- basename
- readlink
- ln
- echo
- sort
- wc
- chmod
- mv
- cat

Of course, we can make a slice for each of them and include them here. But it might be troublesome later to maintain this. It will also look ugly to be honest.

Let me know if you think I should go ahead with just having those individual slices anyway.

Reply:
that's what chisel is for though. leaving more content than what's needed just because it's less ugly ... :shrug: as a user I never see how ugly it is, I just care about the chiselled result.

You may group them if appropriate

Reply:
Sliced coreutils in https://github.com/canonical/chisel-releases/pull/339/commits/28d3b4fe82b5ce3bbb7c95cfd948d729308e5295 and https://github.com/canonical/chisel-releases/pull/339/commits/e35a908ab04e9c721f69f949b67e97b80c05aa0c.
Resolved by https://github.com/canonical/chisel-releases/pull/339/commits/cdbf56e59e05e7cbf97c7b1e3855ea55c83a7073.